util-hadoop/src/main/java/com/google/cloud/hadoop/util/RefreshTokenAuth2Provider.java

```diff
@@ -0,0 +1,230 @@
+package com.google.cloud.hadoop.util;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.google.common.flogger.GoogleLogger;
+import com.google.common.io.BaseEncoding;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.http.HttpHeaders;
+
+public class RefreshTokenAuth2Provider implements AccessTokenProvider {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private static DateTimeFormatter dateFormat =
+      DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS").withZone(ZoneId.systemDefault());
+
+  private static final int CONNECT_TIMEOUT = 30 * 1000;
+  private static final int READ_TIMEOUT = 30 * 1000;
+
+  public static final String CONFIG_REFRESH_TOKEN = "fs.gs.auth.refresh.token";
+  public static final String CONFIG_TOKEN_ENDPOINT = "fs.gs.auth.token.endpoint";
+  public static final String CONFIG_CLIENT_ID = "fs.gs.auth.client.id";
+  public static final String CONFIG_CLIENT_SECRET = "fs.gs.auth.client.secret";
+  public static final String CONFIG_PROXY_ADDRESS = "fs.gs.proxy.address";
+  public static final String CONFIG_PROXY_USERNAME = "fs.gs.proxy.username";
+  public static final String CONFIG_PROXY_PASSWORD = "fs.gs.proxy.password";
```

Comment:
Can we inject these configurations instead of defining statically within the class ?

Reply:
sure thing, my bad